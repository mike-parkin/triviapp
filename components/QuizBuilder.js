import React, { useState, useEffect } from 'react'
import {
    StyleSheet,
    ScrollView,
    View,
    Text,
    Button,
} from 'react-native'
import { StatusBar } from 'expo-status-bar'
import RadioForm, {RadioButton, RadioButtonInput, RadioButtonLabel} from 'react-native-simple-radio-button';
import RNPickerSelect from 'react-native-picker-select'
//api calls
import { getCategories } from '../api/openTDb'

// props fo the radio buttons
const qNumberProps = [
    {label: '5', value: '5'},
    {label: '10', value: '10'},
    {label: '15', value: '15'},
]
const qDifficultyProps = [
    {label: 'Any difficulty', value: ''},
    {label: 'Easy', value: 'easy'},
    {label: 'Medium', value: 'medium'},
    {label: 'Hard', value: 'hard'},
]

const QuizBuilder = ({ navigation }) => {
    const [ categories, setCategories ] = useState([])
    const [form, setForm] = useState({
        num: '10',
        category: '',
        difficulty: ''
    })

    const fetchCategories = () => {
        getCategories()
            .then(res => {
                setCategories(res)
            })
    }

    useEffect(() => {
        fetchCategories()
    }, [])
    
    const catItems = categories.map(category => {
        return { label: category.name, value: category.id }
    })

    return (
        <View style={styles.container}>
            <Text>QuizBuilder</Text>
            <View style={styles.qNumRadioGroup}>
                <Text>How many Questions?</Text>
                <RadioForm
                    radio_props={qNumberProps}
                    initial={1}
                    onPress={value => {setForm({ ...form, num: value})}}
                    formHorizontal={true}
                    labelHorizontal={true}
                    labelStyle={{ paddingRight: 20 }}
                />
            </View>
            <View style={styles.qDifficultyRadioGroup}>
                <Text>Choose the difficulty</Text>
                <RadioForm
                    radio_props={qDifficultyProps}
                    initial={0}
                    onPress={value => {setForm({ ...form, difficulty: value})}}
                    labelHorizontal={true}
                    labelStyle={{ paddingRight: 20 }}
                />
            </View>
            <View style={styles.categorySelector}>
                <Text>Select a difficulty</Text>
                <RNPickerSelect
                    onValueChange={newValue => setForm({...form, category: newValue})}
                    items={[
                        {label: "Any Category", value: ""},
                        ...catItems
                    ]}
                    value={""}
                />
            </View>
            <View>
                <Button
                    title="Create Quiz"
                    onPress={() => navigation.navigate("QuizPage", { form: form })}
                />
            </View>
            <StatusBar style="light" />
        </View>
    )
}

export default QuizBuilder

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    qNumRadioGroup: {
        alignItems: 'center'
    },
    qDifficultyRadioGroup: {
        alignItems: 'center'
    },
    categorySelector: {
        alignItems: 'center',
        marginHorizontal: 'auto',
        marginVertical: '20px',
        width: '70%'
    }
})

