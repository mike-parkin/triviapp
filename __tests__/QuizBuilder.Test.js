import React from 'react'
import renderer from 'react-test-renderer'
import QuizBuilder, { form, setForm } from '../components/QuizBuilder'

// jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

test('renders correctly', async () => {
    const tree = renderer.create(<QuizBuilder />).toJSON()
    expect(tree).toMatchSnapshot()
})

test('Question data changes', async () => {
    const test = 'test'
    expect(test).toEqual('test')
})
