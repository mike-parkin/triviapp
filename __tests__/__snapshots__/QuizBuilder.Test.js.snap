// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text>
    QuizBuilder
  </Text>
  <View>
    <Text>
      How many Questions?
    </Text>
    <View
      accessibilityRole="radiogroup"
      accessible={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "flexDirection": "row",
          }
        }
      >
        <Text>
          5
        </Text>
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="radio"
          accessibilityState={
            Object {
              "checked": false,
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "padding": 6,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View>
        <Text>
          10
        </Text>
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="radio"
          accessibilityState={
            Object {
              "checked": true,
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "padding": 6,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View>
        <Text>
          15
        </Text>
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="radio"
          accessibilityState={
            Object {
              "checked": false,
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "padding": 6,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="ios_touchable_wrapper"
      >
        <View
          pointerEvents="box-only"
        >
          <TextInput
            allowFontScaling={true}
            editable={false}
            rejectResponderTermination={true}
            style={
              Array [
                undefined,
                Object {
                  "color": "#c7c7cd",
                },
              ]
            }
            testID="text_input"
            underlineColorAndroid="transparent"
            value="Select an item..."
          />
        </View>
      </View>
      <Modal
        hardwareAccelerated={false}
        onOrientationChange={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        testID="ios_modal"
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="ios_modal_top"
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "borderTopColor": "#dedede",
                "borderTopWidth": 1,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "zIndex": 2,
              },
              undefined,
            ]
          }
          testID="input_accessory_view"
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#a1a1a1",
                      "borderRightWidth": 1.5,
                      "borderTopWidth": 1.5,
                      "height": 15,
                      "width": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 11,
                      "transform": Array [
                        Object {
                          "translateY": 4,
                        },
                        Object {
                          "rotate": "-45deg",
                        },
                      ],
                    },
                    undefined,
                    Object {},
                  ]
                }
              />
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#a1a1a1",
                      "borderRightWidth": 1.5,
                      "borderTopWidth": 1.5,
                      "height": 15,
                      "width": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 22,
                      "transform": Array [
                        Object {
                          "translateY": -5,
                        },
                        Object {
                          "rotate": "135deg",
                        },
                      ],
                    },
                    undefined,
                    Object {},
                  ]
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 4,
                "left": 4,
                "right": 4,
                "top": 4,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="done_button"
          >
            <View
              testID="needed_for_touchable"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#007aff",
                      "fontSize": 17,
                      "fontWeight": "600",
                      "paddingRight": 11,
                      "paddingTop": 1,
                    },
                    undefined,
                    Object {},
                  ]
                }
                testID="done_text"
              >
                Done
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#d0d4da",
                "justifyContent": "center",
              },
              Object {
                "height": 215,
              },
              undefined,
            ]
          }
        >
          <View>
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "Select an item...",
                    "testID": undefined,
                    "textColor": 4288585892,
                    "value": null,
                  },
                  Object {
                    "label": "test1",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "test1",
                  },
                  Object {
                    "label": "test1",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "test1",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
              testID="ios_picker"
            />
          </View>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
